### DISCUSSION TITLE
Epson L405

### DISCUSSION THREAD

**Discussion by Kennnwxyz** [Ircama/epson_print_conf#4] [Q&A] [answered]
Hello, I'm trying to reset the ink counter on my L405 but seems like it's not on the list
I already searched the similar models = ["ET-2610", "L3060", "L396", "L405"] but none of them are on the list.
Is there any other way to reset the counter or am I doing something wrong?

**Comment by dgerber** [comment]
> Hello, I'm trying to reset the ink counter on my L405 but seems like
> it's not on the list I already searched the similar models =
> ["ET-2610", "L3060", "L396", "L405"] but none of them are on the list.
> Is there any other way to reset the counter or am I doing something
> wrong?

You can find the relevant data here:
https://codeberg.org/atufi/reinkpy/src/branch/main/reinkpy/epson.toml

**Reply by Kennnwxyz** [reply]
it seems like my printer (L405) is not included in the list of valid printer in epson_print_conf.py, so I need to add it myself from https://codeberg.org/atufi/reinkpy/src/branch/main/reinkpy/epson.toml? How can I do that, do you mind teaching me?
![image](https://github.com/Ircama/epson_print_conf/assets/104927126/d3e8a920-2c00-45dc-ae26-5d08b8dc8024)

[[EPSON]]
  | rkey = 0x0395
  | wkey = "Nbojokbv"
  | wkey1 = "Maninjau"
  | models = ["L405"]
  | mem = [
  | { addr = [0x18,0x19] , desc = "Waste counter" },
  | { addr = [0x1C,0x1D,0x2E] , desc = "Waste counters (?)" , reset = [0x00,0x00,0x5E] },
  | { addr = [0x1E] , desc = "Waste counter" },
  | ]
  | rlen = 2
  | wlen = 2
  | mem_high = 0x1FF

**Comment by dgerber** [comment, marked as answer]
> it seems like my printer (L405) is not included in the list of valid printer in epson_print_conf.py, so I need to add it myself from https://codeberg.org/atufi/reinkpy/src/branch/main/reinkpy/epson.toml? How can I do that, do you mind teaching me?
You have two options:

1. Patch and use this tool, with something like this (untested):

```
diff --git a/epson_print_conf.py b/epson_print_conf.py
index db48b24..a9fa33c 100644
--- a/epson_print_conf.py
+++ b/epson_print_conf.py
@@ -82,6 +82,13 @@ class EpsonPrinter:
             "read_key": [65, 9],
             # uncompleted
         },
+        "L405": {
+            "read_key": [0x95, 0x03],
+            "write_key": b'Maninjau',
+            "raw_waste_reset": {
+                0x18:0, 0x19:0, 0x1c:0, 0x1d:0, 0x2e:0x5e, 0x1e:0
+            },
+        },
         "L3250": {
             "read_key": [74, 54],
             "write_key": b'Maribaya',
```

2. Or try with reinkpy. It supports connection via USB, but so far not via network SNMP/UDP/IP
like epson_print_conf does. I've made a GUI for it which should work at least on linux.

**Comment by Kennnwxyz** [comment]
I followed your 1st option and it works, thank you so much!
![image](https://github.com/Ircama/epson_print_conf/assets/104927126/200b70b1-12a7-4ebd-b188-c253a16ee54e)

also verified it by checking the counter using wic utility and it says 0%

**Comment by Ircama** [comment]
L405 added.

**Reply by vmionix** [reply]
Heya, I'm no programmer, but I resetted the waste ink for an L3060, by renaming L405 into L3060 since it had the same properties.

I'm sure there's a lot of these which are just duplicates of eachother, and surely the reinkpy database can be merged into this app

Thanks a lot for your contribution @Ircama , saves me money from going to Epson, and risking my computer to malware/ransomware etc;

**Comment by Ircama** [comment]
L3060 printer added
